File: PiSecuencial.cpp
#include <iostream>
#include <random>
#include <chrono>
#include <ctime>
#include <cstdlib>
using namespace std;


int main()
{
    std::chrono::time_point<std::chrono::system_clock> start, end;

    int nPuntos,aciertos=0;
    double PI = 0.0;

    cout << "Introduzca el numero de puntos a lanzar\n";
    cin >> nPuntos;
    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_real_distribution<> dis(0, 1);

    std::cout << '\n';
    start = std::chrono::system_clock::now();

    for(int i=0;i<nPuntos;i++)
    {
        double x, y;
        x = dis(gen);
        y = dis(gen);

        if((x*x+y*y)<1)
            aciertos++;
    }


    PI = (4.0*aciertos/nPuntos);

    end = std::chrono::system_clock::now();

    cout <<"La aproximacion del numero PI con "<<nPuntos<< " es "<< PI;


    std::chrono::duration<double> elapsed_seconds = end-start;
    std::time_t end_time = std::chrono::system_clock::to_time_t(end);

    std::cout << "\nTiempo: " << elapsed_seconds.count() << "s\n";
    system("pause");
    return(0);

}

File: piSecuencial.java
/**
 * @author Pablo Piedad Garrido
 *  Este programa calcula una aproximacion del numero PI, la cual será más o menos
 * exacta según el número de puntos a lanzar. A mayor numero de puntos, mayor será la exactitud.	
 */

import java.util.Random;
import java.util.Scanner;

public class piSecuencial {

	/**
	 * MAIN: Aqui se pide el numero de puntos, el cual se pasa por valor al
	 * metodo calculoAciertos. Con el número de aciertos y el número de puntos
	 * lanzados podemos calcular la aproximación del número PI.
	 * 
	 * @param args
	 */
	public static void main(String[] args) {
		double pi = 0.0;
		int nPuntos, aciertos;
		Scanner S = new Scanner(System.in);
		System.out.println("Introduzca el numero de puntos a lanzar");
		nPuntos = S.nextInt();
		long inicio = System.nanoTime();
		aciertos = calculoAciertos(nPuntos);

		pi = (4.0 * aciertos / nPuntos);
		long fin = (long) ((System.nanoTime() - inicio) / 1.0e6);
		System.out.println(pi);
		System.out.println("Ha tardado " + fin + "milisegundos");
	}

	/**
	 * Este metodo es el encargado de hallar el numero de aciertos
	 * 
	 * @param nPuntos
	 * @return nAciertos
	 */
	public static int calculoAciertos(int nPuntos) {
		Random r = new Random(System.nanoTime());
		int nAciertos = 0;
		for (int i = 0; i < nPuntos; i++) {
			double cx = r.nextDouble();
			double cy = r.nextDouble();

			if ((Math.pow(cx, 2) + Math.pow(cy, 2)) < 1)
				nAciertos++;
		}

		return nAciertos;
	}
}
